wsidicom/instance.py

```diff
@@ -592,6 +396,132 @@ def slice_thickness(self) -> Optional[float]:
         """Return slice thickness."""
         return self._slice_thickness
 
+    @property
+    def wsi_type(self) -> str:
+        return self._get_wsi_flavor(self.ImageType)
+
+    @classmethod
+    def is_supported_wsi_dicom(
+        cls,
+        dataset: Dataset,
+        transfer_syntax: UID
+    ) -> Optional[str]:
+        """Check if dataset is dicom wsi type and that required attributes
+        (for the function of the library) is available.
+        Warn if attribute listed as requierd in the library or required in the
+        standard is missing.
+
+        Parameters
+        ----------
+        dataset: Dataset
+            Pydicom dataset to check if is a WSI dataset.
+        transfer_syntax: UID
+            Transfer syntax of dataset.
+
+        Returns
+        ----------
+        Optional[str]
+            WSI image flavor
+        """
+
+        sop_class_uid = getattr(dataset, "SOPClassUID", "")
+        if sop_class_uid != WSI_SOP_CLASS_UID:
+            return None
```

Comment:
It would be good if the end-user understands specifically why a file is not validated. Here e.g. the user would want to know that the SOPClassUID is not a WSI.

Reply:
Give warning with what the found sop class was.